File: otherLangs/haskell/06.hs
import Control.Monad ()
import Data.List.Split ( splitOn )
import qualified Data.Set as Set

main :: IO ()
main = do
  contents <- readFile "inputs/06.txt"
  let input = map (map Set.fromList . lines) $ splitOn "\n\n" contents
  print $ sum $ map (Set.size . Set.unions) input
  print $ sum $ map (\(x:xs) -> Set.size $ foldl Set.intersection x xs) input

File: 2020/06.js
import { readFile, sum, intersection } from '../tools.ts';

const input = readFile('inputs/06.txt').split('\n\n');

console.log(sum(input.map(g => new Set(g.split('').filter(c => /[a-z]/.test(c))).size)))
console.log(sum(input.map(g => intersection(...g.split('\n').map(l => new Set(l.split('')))).size)))
